src/test/python/deep_qa_test/layers/attention/gated_attention_test.py

```diff
@@ -0,0 +1,102 @@
+# pylint: disable=no-self-use,invalid-name
```

Comment:
Might be good to have a test for the masked case as well, given I'm not 100% sure it would pass in the addition and concatenation cases.